tests/utils/tracer.py

```diff
@@ -64,14 +64,25 @@ def __init__(self):
         self._update_writer()
 
     def _update_writer(self):
-        self.writer = DummyWriter(
-            hostname=self.writer.api.hostname,
-            port=self.writer.api.port,
-            filters=self.writer._filters,
-            priority_sampler=self.writer._priority_sampler,
-        )
+        if not isinstance(self.writer, DummyWriter):
```

Comment:
Could we add a comment explaining this new block here? I'm not quite sure what's going on or why this is needed

Reply:
Sure I'll add a comment. This was so we can test to see which writer the tracer was created with before it was overridden by the Dummy Writer (since all the other tests use the dummy writer). The second change lets us instantiate the DummyWriter using a Writer without an api property, (like the LogWriter).

Reply:
Gotcha, yeah a comment here for future readers will be great, thanks! 😄